src/main/java/com/limechain/sync/warpsync/SyncedState.java

```diff
@@ -112,7 +112,10 @@ public void syncCommit(CommitMessage commitMessage, PeerId peerId) {
         }
     }
 
-    private void updateState(CommitMessage commitMessage) {
+    private synchronized void updateState(CommitMessage commitMessage) {
```

Comment:
Why should this method be `synchronized`

Reply:
We are receiving commit messages from multiple peers at the same time and they start updating the state at the same time. We were updating the finalized block multiple times for the same block.